analysis/druidrestoration/src/modules/features/Tranquility.ts

```diff
@@ -0,0 +1,49 @@
+import SPELLS from 'common/SPELLS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { EventType, HealEvent, RefreshBuffEvent } from 'parser/core/Events';
+import EventEmitter from 'parser/core/modules/EventEmitter';
+
+/**
+ * **Tranquility**
+ * Heals all allies within 40 yards for [(59% of Spell power) * 5] over 8 sec.
+ * Each heal heals the target for another (6.88% of Spell power) over 8 sec, stacking.
+ * Healing increased by 100% when not in a raid.
+ */
+class Tranquility extends Analyzer {
+  static dependencies = {
+    eventEmitter: EventEmitter,
+  };
+
+  eventEmitter!: EventEmitter;
+
+  constructor(options: Options) {
+    super(options);
+
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell(SPELLS.TRANQUILITY_HEAL),
+      this.onTranqHeal,
+    );
+  }
+
+  /**
+   * When tranquility directly heals it refreshes the HoT, but no refresh buff event is emitted.
+   * This fabricates the event.
+   */
+  onTranqHeal(event: HealEvent) {
```

Comment:
Interesting, not sure how I missed those events the first time. However, observation of buff remove times suggests the Buff Stack Gained events actually *are* refreshing the duration.

Reply:
Yeah. Buff stack events have the power to do that. You should update the hot tracker to handle buff stack events instead of throwing a new fab event imo

Reply:
Yeah you're probably right, I'll make the change